lib/Dialect/TTNN/Pipelines/TTNNPipelines.cpp

```diff
@@ -107,8 +107,21 @@ void createTTNNPipelineDeallocPassFromString(OpPassManager &pm,
   createTTNNPipelineDeallocPass(pm, *optionsStruct);
 }
 
+void createTTNNPipelineTTIRBroadcastFoldPass(
+    OpPassManager &pm, const TTIRToTTNNBackendPipelineOptions &options) {
+  pm.addPass(mlir::tt::ttir::createTTIRBroadcastFold());
+}
+
+void createTTNNPipelineTTIRBroadcastFoldPassFromString(OpPassManager &pm,
+                                                       std::string options) {
+  auto optionsStruct =
+      TTIRToTTNNBackendPipelineOptions::createFromString(options);
+  createTTNNPipelineDeallocPass(pm, *optionsStruct);
+}
+
 void createTTIRToTTNNBackendPipeline(
     OpPassManager &pm, const TTIRToTTNNBackendPipelineOptions &options) {
+  createTTNNPipelineTTIRBroadcastFoldPass(pm, options);
```

Comment:
I think we should use the cannonicalizer that runs between all passes for this. In my PR here: https://github.com/tenstorrent/tt-mlir/pull/1332 I am adding a `TTIRToTTIRDecomposition` pattern that inserts a broadcast. Since with this placement of `TTNNPipelineTTIRBroadcastFoldPass` is before `TTIRToTTIRDecompositonPass` the broadcast will not be removed.

Reply:
Ok I will move the `TTNNPipelineTTIRBroadcastFoldPass` after `TTIRToTTIRDecompositonPass` to catch that case.